src/mpid/ch4/src/ch4_coll.h

```diff
@@ -10,7 +10,28 @@
 #include "ch4_proc.h"
 #include "ch4_coll_impl.h"
 
-MPL_STATIC_INLINE_PREFIX int MPID_Barrier(MPIR_Comm * comm, MPIR_Errflag_t * errflag)
+/*
+=== BEGIN_MPI_T_CVAR_INFO_BLOCK ===
+
+cvars:
+    - name        : MPIR_CVAR_BARRIER_COMPOSITION
+      category    : COLLECTIVE
+      type        : int
+      default     : 0
+      class       : none
+      verbosity   : MPI_T_VERBOSITY_USER_BASIC
+      scope       : MPI_T_SCOPE_ALL_EQ
+      description : >-
+        Select composition (inter_node + intra_node) for Barrier
+        0 Auto selection
+        1 NM + SHM
+        2 NM only
```

Comment:
Ahh, that makes sense. Why don't we change `_alpha` and `_beta` from the algorithm names to reflect this semantic directly?

Reply:
Also, instead of int, why don't we use an enum so we can use the semantic meaningful algorithm name, e.g. "auto, nm_only, nm_and_shm"?

Reply:
yeah, I am not sure which way is better. I think the original author borrowed a scheme from MVAPICH. the little complexity here is that even nm_and_shm may have two variants in bcast. It is clumsy to describe in the name string. And hopefully the user never have to worry about using this CVAR (json file should take care of it). This is probably only for an experienced user or for testing/autotuning purpose.
I agree we can discuss for a better scheme later, and we can re-visit this issue after we have all the pieces merged and the clean up would be more straightforward at that time.

Reply:
It doesn't need to be perfect, but some partial semantic names will be better than `alpha, beta, ...`, or `0, 1, 2`. It is not just for users. Mostly is for the sanity of maintainers, who need to understand the code without navigating the cryptic chains.

Reply:
I understand the concern. But to some extent the comment block above already explain the meaning of 0, 1, and 2. So it is not totally confusing. But a better scheme would certainly be desirable. If this is not urgent, we can revisit this CVAR naming scheme once we are ready.